File: leetcode/1662. Check If Two String Arrays are Equivalent/main.py
class Solution:
    def arrayStringsAreEqual(self, word1: List[str], word2: List[str]) -> bool:
        for ch1, ch2 in zip(self.__generate_ch(word1), self.__generate_ch(word2)):
            if ch1 != ch2:
                return False
        return True

    def __generate_ch(self, words: List[str]):
        for word in words:
            for ch in word:
                yield ch
        yield None

File: leetcode/1662. Check If Two String Arrays are Equivalent/main.go
// Solved by @kitanoyoru
// https://leetcode.com/problems/check-if-two-string-arrays-are-equivalent/

package main

import "strings"

func concArrOfString(a []string) string {
	var b string
	for _, v := range a {
		b += v
	}

	return b
}

func arrayStringsAreEqual(word1 []string, word2 []string) bool {
	a := concArrOfString(word1)
	b := concArrOfString(word2)

	return strings.EqualFold(a, b)
}
